services/debugger/destination/cache.go

```diff
@@ -0,0 +1,108 @@
+package destinationdebugger
+
+import (
+	"fmt"
+	"github.com/rudderlabs/rudder-server/config"
+	"sync"
+	"time"
+)
+
+var (
+	CacheTTL     time.Duration
+	CacheMaxSize int
+)
+
+func loadCacheConfig() {
+	config.RegisterIntConfigVariable(2, &CacheMaxSize, true, 1, "DestinationDebugger.maxEventsCacheSize")
+	config.RegisterDurationConfigVariable(720 * time.Hour, &CacheTTL, true, 1, "DestinationDebugger.maxEventsCacheTTL")
+}
+
+type cacheItem struct {
+	objs       []*DeliveryStatusT
+	lastAccess time.Time
+}
+
+type Cache struct {
+	lock sync.RWMutex
+	Size   int
+	KeyTTL time.Duration
+	m      map[string]*cacheItem
+	once sync.Once
+}
+
+func (c *Cache) init() {
+	c.once.Do(func() {
+		c.Size = CacheMaxSize
```

Comment:
Why do we need to store CacheMaxSize in Cache Struct. If we store it this way , configuring this variable wont be hot reloadable

Reply:
fixed